modular/manager/manager.go

```diff
@@ -1195,3 +1201,24 @@ func (m *ManageModular) QueryRecoverProcess(ctx context.Context, vgfID, gvgID ui
 	flag := m.recoverProcessCount.Load() > 0 || m.verifyTerminationSignal.Load() > 0
 	return res, flag, nil
 }
+
+func (m *ManageModular) syncAvailableVGF(ctx context.Context) {
+	var (
+		err error
+		sp  *sptypes.StorageProvider
+	)
+
+	// query meta
+	if sp, err = m.baseApp.Consensus().QuerySPByID(context.Background(), m.spID); err != nil {
+		log.CtxErrorw(ctx, "failed to list sps", "error", err)
+		return
+	}
+
+	// only pick vgf when sp is STATUS_IN_SERVICE
```

Comment:
do we need to take GVGPreferSPList and SPBlackList into account when picking vgf?

Reply:
SPBlackList will be used when picking VGF
![image](https://github.com/bnb-chain/greenfield-storage-provider/assets/122767193/87f17df1-5c6a-4fee-8209-816d009f2e56)

GVGPreferSPList will be used when creating GVG
<img width="1160" alt="image" src="https://github.com/bnb-chain/greenfield-storage-provider/assets/122767193/aae2f04d-b873-49e0-9ff9-1aecb646b20f">

![image](https://github.com/bnb-chain/greenfield-storage-provider/assets/122767193/7922d52f-679e-487e-a6e1-11cf3fb43805)